### PR TITLE
Allow users to update restore namespaces

### PR DESCRIPTION
**Pull request by harsh-px**
**Does this PR need a release note?:** yes

```release-note
Allow users to update the restore namespaces for a group volume snapshot after it was created
```

Signed-off-by: Harsh Desai <harsh@portworx.com>

### CODE REVIEW

File: pkg/groupsnapshot/controllers/groupsnapshot.go [hunk 1]
```diff
@@ -569,7 +569,44 @@ func (m *GroupSnapshotController) handlePostSnap(groupSnap *stork_api.GroupVolum
...
+		}
+	}
+
+	if latestRestoreNamespacesInCSV != currentRestoreNamespaces {
+		log.GroupSnapshotLog(groupSnap).Infof("Updating restore namespaces for groupsnapshot to: %s",
+			latestRestoreNamespacesInCSV)
+		for _, childSnap := range childSnapshots {
+			vs, err := k8s.Instance().GetSnapshot(childSnap.VolumeSnapshotName, groupSnap.GetNamespace())
+			if err != nil {
+				return err
```
Comment (disrani-px) [inline pkg/groupsnapshot/controllers/groupsnapshot.go]: This should ignore snapshots that might have been deleted

File: pkg/groupsnapshot/controllers/groupsnapshot.go [hunk 2]
```diff
@@ -569,7 +569,44 @@ func (m *GroupSnapshotController) handlePostSnap(groupSnap *stork_api.GroupVolum
 }
 
 func (m *GroupSnapshotController) handleFinal(groupSnap *stork_api.GroupVolumeSnapshot) error {
-	// currently nothing to do
+	// Check if user has updated restore namespace
+	latestRestoreNamespacesInCSV := strings.Join(groupSnap.Spec.RestoreNamespaces, ",")
```
Comment (disrani-px) [inline pkg/groupsnapshot/controllers/groupsnapshot.go]: Move inside the if on line 577.

File: pkg/groupsnapshot/controllers/groupsnapshot.go [hunk 3]
```diff
@@ -569,7 +569,44 @@ func (m *GroupSnapshotController) handlePostSnap(groupSnap *stork_api.GroupVolum
...
+			return err
+		}
+
+		childSnapAnnotations := vsObject.Metadata.Annotations
+		if childSnapAnnotations != nil {
+			currentRestoreNamespaces = childSnapAnnotations[snapshot.StorkSnapshotRestoreNamespacesAnnotation]
+		}
+	}
+
+	if latestRestoreNamespacesInCSV != currentRestoreNamespaces {
```
Comment (disrani-px) [inline pkg/groupsnapshot/controllers/groupsnapshot.go]: From here to the end of the function is required only if snapshots are present, so move in the scope of the if on line 577.

File: pkg/groupsnapshot/controllers/groupsnapshot.go (hunk 2 above)
Comment (harsh-px) [reply pkg/groupsnapshot/controllers/groupsnapshot.go]: done

File: pkg/groupsnapshot/controllers/groupsnapshot.go (hunk 3 above)
Comment (harsh-px) [reply pkg/groupsnapshot/controllers/groupsnapshot.go]: done

File: pkg/groupsnapshot/controllers/groupsnapshot.go (hunk 1 above)
Comment (harsh-px) [reply pkg/groupsnapshot/controllers/groupsnapshot.go]: done

**Comment by harsh-px** [comment]
@disrani-px what is the branch for 2.1.1 milestone ?

**Pull request merged**